keydata.go

```diff
@@ -216,6 +327,30 @@ func (d *keyData) Unmarshal(r io.Reader) (nbytes int, err error) {
 			authModeHint:      raw.AuthModeHint,
 			staticPolicyData:  raw.StaticPolicyData.data(),
 			dynamicPolicyData: raw.DynamicPolicyData.data()}
+	case 1:
+		var splitData afSplitDataRaw
+		n, err := tpm2.UnmarshalFromReader(r, &splitData)
+		nbytes += n
+		if err != nil {
+			return nbytes, xerrors.Errorf("cannot unmarshal split data: %w", err)
```

Comment:
Should these error cases `return` `0` or `nbytes`? The `d.version` code in `keydata.go` appeared to `return 0, ...` on error.

Reply:
The top-level API's in go-tpm2 return the number of bytes read, even in the case of an error (see https://godoc.org/github.com/canonical/go-tpm2/mu#UnmarshalFromReader), which is why nbytes is returned here. The case where 0 is returned is probably a mistake, but I've just pushed some changes to go-tpm2 to avoid the need for this logic to be here at all:

https://github.com/canonical/go-tpm2/commit/bdbb15b0f6937b4f68708bb533482c144c5071e2